nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/registry/flow/ImportAllFlows.java

```diff
@@ -0,0 +1,298 @@
+    private static final String IMPORT_COMPLETED = "Import completed...";
+    private static final String ALL_BUCKETS_COLLECTED = "All buckets collected...";
+    private static final String ALL_FLOWS_COLLECTED = "All flows collected...";
+    private static final String ALL_FLOW_VERSIONS_COLLECTED = "All flow versions collected...";
+    private final ListBuckets listBuckets;
+    private final ListFlows listFlows;
+    private final ListFlowVersions listFlowVersions;
+    private final ImportFlowVersion importFlowVersion;
+
+    public ImportAllFlows() {
+        super("import-all-flows", StringResult.class);
+        this.listBuckets = new ListBuckets();
+        this.listFlows = new ListFlows();
+        this.listFlowVersions = new ListFlowVersions();
+        this.importFlowVersion = new ImportFlowVersion();
+    }
+
+    @Override
+    protected void doInitialize(Context context) {
+        addOption(CommandOption.INPUT_SOURCE.createOption());
+        addOption(CommandOption.SKIP.createOption());
+
+        listBuckets.initialize(context);
+        listFlows.initialize(context);
+        listFlowVersions.initialize(context);
+        importFlowVersion.initialize(context);
+    }
+
+    @Override
+    public String getDescription() {
+        return "From a provided directory as input, the directory content must be generated by the export-all-flows command, " +
+                "based on the file contents, the corresponding buckets, flows and flow versions will be created." +
+                "If not configured otherwise, already existing objects will be skipped.";
+    }
+
+    @Override
+    public StringResult doExecute(final NiFiRegistryClient client, final Properties properties) throws IOException, NiFiRegistryException, ParseException, CommandException {
+        final boolean skip = Boolean.parseBoolean(getRequiredArg(properties, CommandOption.SKIP));
+        final boolean isInteractive = getContext().isInteractive();
+
+        //Gather all buckets and create a map for easier search by bucket name
+        final Map<String, String> bucketMap = getBucketMap(client, isInteractive);
+
+        // Gather all flows and create a map for easier search by flow name.
+        // As flow name is only unique within the same bucket we need to use the bucket id in the key as well
+        final Map<Pair<String, String>, String> flowMap = getFlowMap(client, bucketMap, isInteractive);
+        final Map<Pair<String, String>, String> flowCreated = new HashMap<>();
+
+        // Gather all flow versions and create a map for easier search by flow id
+        final Map<String, List<Integer>> versionMap = getVersionMap(client, flowMap, isInteractive);
+
+        // Create file path list
+        final List<String> files = getFilePathList(properties);
+
+        // Deserialize file content
+        final List<ImportedSnapshot> importedSnapshots = deserializeSnapshots(files);
```

Comment:
Instead of serializing all snapshots, is it possible to iterate through each one? It looks like that could create problems for sorting, so perhaps there is no other way to avoid loading everything.

Reply:
I'm wondering if we can change the filenames that are produced by export to contain the 3 fields below that we are sorting on, then we can sort the `List<String> files` in the same order?

So exported filenames would be:
```
<Bucket-Name>-<Flow-Name>-<Version>.json
```
This would also have the added benefit of being human readable for anyone looking at the exported files.